grpc/README.md

```diff
@@ -0,0 +1,333 @@
+# gRPC on Apigee
+
+This sample shows how to use Apigee in front of your gRPC backends.
+For this example we will be using Cloud Run to host a gRPC server and configure it as a target server in Apigee. Apigee will act as a simple reverse proxy in front of the gRPC backend. For more detailed information on using gRPC proxies in Apigee, please refer to the official [documentation](https://cloud.google.com/apigee/docs/api-platform/fundamentals/build-simple-api-proxy#creating-grpc-api-proxies).
+
+
+
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Access to deploy proxies, create products, apps and developers in Apigee
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * unzip
+    * curl
+    * jq
+    * npm
+    * docker
+5. Make sure that the following GCP APIs are enabled:
+    * Cloud Run API
+    * Container Registry API
+
+6. Make sure that you have the following permissions:
+
+    * [Artifact Registry Repository Administrator](https://cloud.google.com/iam/docs/understanding-roles#artifactregistry.repoAdmin) role  (roles/artifactregistry.repoAdmin)
+    * [Cloud Run Developer](https://cloud.google.com/run/docs/reference/iam/roles) role (roles/run.developer)
+
+
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=grpc/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+## Create a gRPC backend
```

Comment:
After going through this I think we could make this easier.  Can we pre-build and host a Docker image with the port number change already made?  The Cloud Run deploy step can just reference that image.